### PR TITLE
Remove global slim timeout during testing

### LINKED ISSUE #657: slim.timeout default value
The current default value for Slim timeout is 10 seconds. This is quite short, can this be set to 30 seconds or a minute?

### PR DESCRIPTION
**Pull request by six42**
This fixes #657 #656 #653 
also fixes a bug with multiple arguments in SLIM_FLAGS.

I started preparing automatic test cases. They will follow shortly.

### CODE REVIEW

**Comment by amolenaar** [comment]
I created a few acceptance tests in the timeout branch on my fork. That might be useful.

**Comment by amolenaar** [comment]
Are you limiting the slim.timeout property to only be used while loading the test system?
I would expect it to be used when executing tests as well.

**Comment by six42** [comment]
I exected this as well. But testing shows it is not the case. There is a workaround using SLIM_FLAGS. But also then the timeout doesn't seems to be useful as every testcase will fail afterwards.
See HowtoEnforceA_TimelimitPerTestCommand

I added the usage of slim.timeout while waiting for the SlimHeader message. The remaing usage of slim.timeout is unchanged.

**Comment by six42** [comment]
Fixed the merge conflict to allow automatic merge.

File: src/fitnesse/testsystems/slim/SlimClientBuilder.java [hunk 1]
```diff
@@ -92,9 +94,10 @@ protected String defaultTestRunner() {
     	arguments = ArrayUtils.add(arguments, "-ssl");
     	arguments = ArrayUtils.add(arguments, useSSL);
     }    	
-    String slimFlags = getSlimFlags();
-    if (!"".equals(slimFlags))
-    	arguments = ArrayUtils.add(arguments, slimFlags);
+    String slimFlags[] = getSlimFlags();
+    if (slimFlags != null)
+    	for (String flag : slimFlags)
+    		arguments = ArrayUtils.add(arguments, flag);
```
Comment (amolenaar) [inline src/fitnesse/testsystems/slim/SlimClientBuilder.java]: The method ClientBuilder.parseCommandLine() could be used to parse the flags. That will take into account flags containing quotes.

File: src/fitnesse/testsystems/slim/SlimClientBuilder.java [hunk 2]
```diff
@@ -179,12 +182,12 @@ protected String determineSlimHost() {
...
   }
 
-  protected String getSlimFlags() {
+  protected String[] getSlimFlags() {
     String slimFlags = getVariable("slim.flags");
     if (slimFlags == null) {
       slimFlags = getVariable(SLIM_FLAGS);
     }
-    return slimFlags == null ? "" : slimFlags;
+    return slimFlags == null ? new String[] {} : slimFlags.trim().split("\\s");
```
Comment (amolenaar) [inline src/fitnesse/testsystems/slim/SlimClientBuilder.java]: There is a smarter parser in ClientBuilder.parseCommandLine(). That will take into account quoted values.

Comment (six42) [inline src/fitnesse/testsystems/slim/SlimClientBuilder.java]: Thanks for the hint. Changed it.

**Pull request merged**